python/pyiceberg/schema.py

```diff
@@ -795,3 +796,150 @@ def map(self, map_type: MapType, key_result: Callable[[], IcebergType], value_re
 
     def primitive(self, primitive: PrimitiveType) -> PrimitiveType:
         return primitive
+
+
+def prune_columns(schema: Schema, selected: Set[int], select_full_types: bool) -> Schema:
```

Comment:
Default `select_full_types=True`? That, or I think this would be an internal method. Callers won't know what `select_full_types` means.

Reply:
I've added the default. I wanted to create a new file for the `TableScan` and then import `prune_columns` from `schema`.